source/common/tracing/zipkin/zipkin_tracer_impl.cc

```diff
@@ -94,24 +117,8 @@ Tracing::SpanPtr Driver::startSpan(Http::HeaderMap& request_headers, const std::
     new_zipkin_span = tracer.startSpan(request_headers.Host()->value().c_str(), start_time);
   }
 
-  // Set the trace-id and span-id headers properly, based on the newly-created span structure.
-  request_headers.insertXB3TraceId().value(new_zipkin_span->traceIdAsHexString());
-  request_headers.insertXB3SpanId().value(new_zipkin_span->idAsHexString());
-
-  // Set the parent-span header properly, based on the newly-created span structure.
-  if (new_zipkin_span->isSetParentId()) {
-    request_headers.insertXB3ParentSpanId().value(new_zipkin_span->parentIdAsHexString());
-  }
-
-  // Set the sampled header.
-  request_headers.insertXB3Sampled().value(ZipkinCoreConstants::get().ALWAYS_SAMPLE);
-
-  // Set the ot-span-context header with the new context.
-  SpanContext new_span_context(*new_zipkin_span);
-  request_headers.insertOtSpanContext().value(new_span_context.serializeToString());
-
   ZipkinSpanPtr active_span;
```

Comment:
minor, ZipkinSpanPtr active_span(new ZipkinSpan(*new_zipkin_span, tracer));
and remove reset.